classes/cron_manager.php

```diff
@@ -63,39 +62,29 @@ function() use ($profiler, $timer) {
      *
      * We then check for a new task with the current sample.
      *
-     * @param \ExcimerProfiler $profiler
+     * @param object $profiler
      */
-    public static function on_interval(\ExcimerProfiler $profiler): void {
-        $log = $profiler->flush();
+    public static function on_interval(object $profiler): void {
+        $log = $profiler->profiler->flush();
 
-        $log->rewind();
-        if (!$log->valid()) {
-            debugging('on_interval called with no profile');
-            return;
-        }
-        $sample = $log->current();
-        $log->next();
-        if ($log->valid()) {
-            debugging('extra profile sample found in on_interval');
-        }
-
-        $taskname = self::findtaskname($sample);
-        $currenttime = microtime(true);
+        foreach ($log as $sample) {
```

Comment:
It looks like this would significantly increase the computation time on every interval, and compounding. Would it be possible to store the index previously accessed, and then resume from that point in log entries (assuming the history of the log is immutable)

Reply:
I also think this should be unit tested with the conditions that were causing the problem

Reply:
Calling 'flush()' empties the profiler of its logs, so there is no re-processing.